fbw-a32nx/src/wasm/systems/a320_systems/src/pneumatic.rs

```diff
@@ -1405,8 +1789,62 @@ impl SimulationElement for CrossBleedValve {
     }
 }
 
+struct FlightPhaseLoop {
+    is_climb_active: bool,
+    is_hold_active: bool,
+
+    vertical_speed_greater_140: DelayedTrueLogicGate,
+    vertical_speed_less_80: DelayedTrueLogicGate,
+    vertical_speed_within_140: DelayedTrueLogicGate,
+}
+impl FlightPhaseLoop {
+    pub fn new() -> Self {
+        Self {
+            is_climb_active: false,
+            is_hold_active: false,
+
+            vertical_speed_greater_140: DelayedTrueLogicGate::new(Duration::from_secs(10)),
+            vertical_speed_less_80: DelayedTrueLogicGate::new(Duration::from_secs(10)),
+            vertical_speed_within_140: DelayedTrueLogicGate::new(Duration::from_secs(10)),
+        }
+    }
+
+    pub fn update(&mut self, context: &UpdateContext) {
+        // TODO: Use correct signals here
```

Comment:
You want to implement this in a future PR?

Reply:
Yeah, need some kind of EIU implementation for it.